pkg/event/target/queuestore.go

```diff
@@ -0,0 +1,240 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package target
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/minio/minio/pkg/event"
+)
+
+const (
+	maxLimit = 10000 // Max store limit.
+	eventExt = ".event"
+)
+
+type QueueStore struct {
+	sync.RWMutex
+	directory string
+	eC        uint16
+	limit     uint16
+}
+
+// NewQueueStore - Filestore for persisting events.
+func NewQueueStore(directory string, limit uint16) *QueueStore {
+	if limit == 0 {
+		limit = maxLimit
+	}
+	queueStore := &QueueStore{
+		directory: directory,
+		limit:     limit,
+	}
+	return queueStore
+}
+
+// Staging directory.
+func tmpPath(directory string) string {
+	return directory + "/tmp"
```

Comment:
lets call this `/tmp` as `/mqtt` ? - also no need to have this function just use `filepath.Join(directory, "mqtt")`

Reply:
We cannot rename it to `mqtt` since queueStore is generic here for all the targets

Reply:
It should be configurable, it cannot be the same for all targets @Praveenrajmani . Also /tmp is not meaningful here we should directly use the queueDir provided by the user for the given target in config.json - there is no reason to have another sub-directory inside it.